client.js

```diff
@@ -0,0 +1,198 @@
+'use strict';
+
+const { MTProto } = require('@mtproto/core');
+const {
+  addUsersToMessages,
+  getFormedMessagesFromUpdates,
+  STORAGE_AUTH_KEYS,
+} = require('./helpers');
+
+module.exports = class TelegramClient {
+  constructor({ apiId, apiHash, test }) {
+    this.apiID = apiId;
+    this.apiHash = apiHash;
+    this.test = test;
+    this.channelMessageSubscribers = {};
+  }
+
+  connect(authKeys) {
+    if (this.client) return;
+
+    this.client = new MTProto({
+      api_id: this.apiID,
+      api_hash: this.apiHash,
+      test: this.test,
+    });
+
+    this.client.updates.on('updates', ({ updates, users }) => {
+      const messages = getFormedMessagesFromUpdates({ updates, users });
+      Object.values(messages).forEach(message => {
+        const channelId = message.channelId;
+        if (this.channelMessageSubscribers[channelId]) this.channelMessageSubscribers[channelId](message);
+      });
+    });
+
+    if (!this.hasAuthKeys() && authKeys) this.importAuthKeys(authKeys);
+  }
+
+  disconnect() {
+    Object.values(this.client.rpcs).forEach(({ transport }) => {
+      transport.destroy();
+    });
+    delete this.client;
+  }
+
+  importAuthKeys(authKeys) {
+    if (this.client.storage._prefix === undefined) this.client.storage.setPrefix('');
+    authKeys = JSON.parse(authKeys);
+    for (const key of STORAGE_AUTH_KEYS) {
+      this.client.storage.pSet(key, JSON.parse(authKeys[key]));
+    }
+  }
+
+  getAuthKeys() {
+    const authKeys = {};
+    for (const key of STORAGE_AUTH_KEYS) {
+      authKeys[key] = JSON.stringify(this.client.storage.get(key));
+    }
+    return JSON.stringify(authKeys);
+  }
+
+  hasAuthKeys() {
+    return STORAGE_AUTH_KEYS.every(key => !!this.client.storage.get(key));
+  }
+
+  subscribeToChannelMessages({ channelId, callback }) {
+    this.channelMessageSubscribers[channelId] = callback;
+  }
+
+  unsubscribeFromChannelMessage({ channelId }) {
+    delete this.channelMessageSubscribers[channelId];
+  }
+
+  request(...args) {
+    if (!this.client) throw new Error('No telegram Client');
+
+    return this.client
+      .call(...args)
+      .catch(error => {
+        if (error.error_message.includes('_MIGRATE_')) {
+          const [_, nextDcId] = error.error_message.split('_MIGRATE_'); //eslint-disable-line
+          this.client.setDefaultDc(Number(nextDcId));
+          return this.request(...args);
+        }
+        if (error.error_message === 'AUTH_RESTART') {
+          throw new Error('Your Telegram auth keys are no longer valid, please relink your account');
+        }
+        throw new Error(error.error_message);
+      });
+  }
+
+  getAccountTTL(){
+    return this.request('account.getAccountTTL', {});
+  }
+
+  sendAuthCode({mobile}){
```

Comment:
```suggestion
  startLogin(){
```

also I think phone number should be take at instantiation time